packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -172,11 +172,7 @@ export class SelectedNetworkController extends BaseController<
             path[0] === 'subjects' && path[1] !== undefined;
           if (isChangingSubject && typeof path[1] === 'string') {
             const domain = path[1];
-            if (
-              op === 'add' &&
-              this.state.domains[domain] === undefined &&
-              this.#useRequestQueuePreference
-            ) {
+            if (op === 'add' && this.state.domains[domain] === undefined) {
```

Comment:
incidentally I think the 'remove'  op case here (in the else if) could be guarded with a `useRequestQueuePreference` check since when the setting is toggled off we loop through and run `unsetNetworkClientIdForDomain` against all the domains already. So when a domain's permissions are revoked and the useRequestQueue toggle is off the domain should already be deleted from the `domains` state and the proxy already pointed to the NetworkController's proxy. But the redundancy here isn't hurting anything

Reply:
I'm leaning towards leaving this guard out for now for reasons we discussed privately. If you feel strongly, I can add it back

Reply:
Sorry so just to be clear it wouldn't be adding it back. It would be moving the guard to the `remove` op case. But I think its fine to leave out for now either way